test/mpi/pt2pt/sendself_complex.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpi.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include "mpitest.h"
+#include <assert.h>
+#include <complex.h>
+
+
+/*
+  This test captures the following scenario that is useful for the application HACC:
+  1) Send a complex double data type
+  2) beyond a certain array length (>=4350) and
+  3) when communicating with itself (source = destination)
```

Comment:
Wouldn't the existing `sendrecv1` with `-type=MPI_DOUBLE_COMPLEX` cover it?

Reply:
It seems `sendrecv1` may not support `MPI_DOUBLE_COMPLEX`. The argument is named as `basic_type` and during a quick test, I saw error related to invalid datatype. 
```
assertion array_of_types[i] != MPI_DATATYPE_NULL failed in function DTPI_parse_base_type_str, line 121
Abort(200969219) on node 0 (rank 0 in comm 0): Fatal error in internal_Type_get_extent: Invalid datatype, error stack:
internal_Type_get_extent(72): MPI_Type_get_extent(INVALID DATATYPE, lb=0x7ffd0db20de8, extent=0x1f10f60) failed
internal_Type_get_extent(42): Invalid datatype
```

Reply:
I see. That can be easily fixed.

Reply:
Oh, you need used `MPI_C_DOUBLE_COMPLEX`.

Reply:
The datatype problem got resolved with that. However, for testcase in this PR, a single rank has to act as sender and receiver. So, `sendself`, which is a varialt of `sendrecv1`, may be a better fit. A test string like the following can possibly replace this patch: 
`pt2pt/sendself -type=MPI_C_DOUBLE_COMPLEX -sendcnt=1 -recvcnt=1 -seed=47 -testsize=8192`

Reply:
Added a patch. Kept the original patch (+revert) for reference during the review, will remove later.